src/promptflow-evals/promptflow/evals/synthetic/README.md

```diff
@@ -1,66 +1,280 @@
-# QA.py Usage Guide
+# Usage Guide
 
-This document provides instructions on how to use `qa.py`.
+This document provides instructions on how to use the adversarial simulator.
 
 ## Prerequisites
 
 Ensure you have Python installed on your system. If not, you can download it from [here](https://www.python.org/downloads/).
 
 ## Installation
 
-Install the `promptflow-evals` extra while installing promptflow.
+Install the `promptflow-evals` package.
+
+`pip install promptflow-evals`
 
 ## Usage
-Set the values for `AZURE_OPENAI_API_KEY` and `AZURE_OPENAI_ENDPOINT` in the env or update this script:
 
-```
-import os
-from promptflow.evals.synthetic.qa import QADataGenerator, QAType, OutputStructure
+```python
+from promptflow.evals.synthetic import AdversarialSimulator
+import config
+from azure.identity import DefaultAzureCredential
+from typing import Any, Dict, List, Optional
+import asyncio
+
+
 
-os.environ["AZURE_OPENAI_API_KEY"] = ""
-os.environ['AZURE_OPENAI_ENDPOINT'] = ""
+project_scope = {
+    "subscription_id": config.sub,
+    "resource_group_name": config.rg,
+    "workspace_name": config.project_name,
+    "credential": DefaultAzureCredential(),
+}
 
-model_name = "gpt-4"
+async def callback(
+    messages: List[Dict],
+    stream: bool = False,
+    session_state: Any = None,
+    context: Dict[str, Any] = None
+) -> dict:
+    question = messages["messages"][0]["content"]
+    context = None
+    if 'file_content' in messages["template_parameters"]:
+        question += messages["template_parameters"]['file_content']
+    from openai import AsyncAzureOpenAI
 
-model_config = dict(
-    deployment=model_name,
-    model=model_name,
-    max_tokens=2000,
+    oai_client = AsyncAzureOpenAI(
+        api_key=config.api_key,
+        azure_endpoint=config.endpoint,
+        api_version="2023-12-01-preview",
+    )
+    try:
+        response_from_acs = await oai_client.chat.completions.create(messages=[{"content": question, "role": "user"}], model="gpt-4", max_tokens=300)

```

Comment:
Seems there's no good UI experience of this callback support now. If user only want to define the LLM client, can user just set a model_config or connection instead of define the whole callback function? I guess that will make a more align experience with UI?

Reply:
also, why the variable named as "response_from_acs" , what's the acs here?

Reply:
updated the variable name in readme.
Changing it to support only connections would make it extremely hard to provide samples and support our existing customers and their use cases of the way they use the simulator. 

@minthigpen @luigiw can we table this for the next update post //build?

Reply:
I'm not against to have the callback support here. Actually this provide better flexibility for user to define their own call actions. While I think it's better to also provide a way to let them skip define callback function when they don't want to do that.

Reply:
Yeah, we can probably support that post build.